File: backend/app/schemas/employment.py
# backend/app/schemas/employment.py
from pydantic import BaseModel, Field


class SchoolResponse(BaseModel):
    id: str
    name: str
    slug: str
    code: str | None = None
    report_count: int = 0
    major_count: int = 0


class EmploymentRecordResponse(BaseModel):
    year: int
    degree: str
    total_graduates: int | None
    rates: dict
    employer_ranking: list
    industry_distribution: dict
    destination_region: dict
    school_for_further_study: list


class TrendResponse(BaseModel):
    years: list[int]
    employment_rate: list[float | None]
    further_study_rate: list[float | None]
    civil_service_rate: list[float | None]
    abroad_rate: list[float | None]


class EmploymentSearchResponse(BaseModel):
    school: SchoolResponse | None
    major: str | None
    records: list[EmploymentRecordResponse]
    trend: TrendResponse | None


class EmploymentStatsResponse(BaseModel):
    school_count: int
    report_count: int
    major_count: int
    year_range: tuple[int | None, int | None]


class SearchBody(BaseModel):
    """POST /api/employment/search 的请求体。

    从 API 路由文件迁移至 schemas，保证请求/响应模型统一在 schemas 层维护。
    """

    school: str = Field(max_length=200)
    major: str
    year: int | None = None
    degree: str | None = None


class MajorQuery(BaseModel):
    """POST /api/employment/majors 的请求体。"""

    school: str = Field(max_length=200)

class CitySalaryBand(BaseModel):
    """城市岗位薪资带（B4 就业市场概览，min/max 为样本极值）。"""

    city: str
    sample_count: int
    min_k: int | None = None
    max_k: int | None = None


class SchoolEmploymentSample(BaseModel):
    """院校就业率样本——只列库内真实有值的院校，覆盖度如实（B4）。"""

    name: str
    employment_rate: float | None = None
    grad_school_rate: float | None = None
    report_index_url: str | None = None


class MarketOverviewResponse(BaseModel):
    """GET /api/employment/market-overview（B4）：companies/salary 真实聚合。"""

    company_total: int
    top_industries: list[dict]
    salary_total: int
    city_salary_bands: list[CitySalaryBand]
    school_employment_samples: list[SchoolEmploymentSample]
    school_employment_coverage: int
    school_total: int

File: frontend/lib/api/employment.ts
import type {
  EmploymentSearchResult,
  SchoolInfo,
  EmploymentStats,
  CommunitySubmit,
  CommunityReport,
  CommunityAggregate,
  CommunityStats,
  InterviewSubmit,
  InterviewReport,
  InterviewAggregate,
  InterviewStats,
  CompanyInfo,
  Company,
  PaginatedResponse,
} from "@/types";
import { request, buildQuery } from "./client";

// ===== 就业数据搜索 =====
export const employmentApi = {
  search: (params: {
    school: string;
    major: string;
    year?: number;
    degree?: string;
  }) =>
    request<EmploymentSearchResult>("/api/employment/search", {
      method: "POST",
      body: JSON.stringify(params),
    }),

  schools: () => request<SchoolInfo[]>("/api/employment/schools"),

  majors: (school: string) =>
    request<string[]>("/api/employment/majors", {
      method: "POST",
      body: JSON.stringify({ school }),
    }),

  stats: () => request<EmploymentStats>("/api/employment/stats"),

  // 批量获取公司元数据（消除前端 N+1 调用）
  batchCompanies: (ids: string[]) =>
    request<Company[]>("/api/employment/companies/batch", {
      method: "POST",
      body: JSON.stringify({ ids }),
    }),

  // 就业市场概览（B4，2026-09-26）：companies/salary_benchmarks 真实聚合
  marketOverview: () =>
    request<MarketOverview>("/api/employment/market-overview"),
};

export interface MarketOverview {
  company_total: number;
  top_industries: { industry: string; count: number }[];
  salary_total: number;
  city_salary_bands: { city: string; sample_count: number; min_k: number | null; max_k: number | null }[];
  school_employment_samples: {
    name: string;
    employment_rate: number | null;
    grad_school_rate: number | null;
    report_index_url: string | null;
  }[];
  school_employment_coverage: number;
  school_total: number;
}

// ===== 社区数据 =====
export const communityApi = {
  submit: (body: CommunitySubmit) =>
    request<CommunityReport>("/api/community/submit", {
      method: "POST",
      body: JSON.stringify(body),
    }),
  myReports: (params?: { page?: number; page_size?: number }) =>
    request<PaginatedResponse<CommunityReport>>(
      `/api/community/my-reports${buildQuery((params as Record<string, string | undefined | null>) || {})}`,
    ),
  remove: (id: string) =>
    request<void>(`/api/community/${id}`, { method: "DELETE" }),
  aggregate: (body: { school: string; major: string; year?: number }) =>
    request<CommunityAggregate>("/api/community/aggregate", {
      method: "POST",
      body: JSON.stringify(body),
    }),
  stats: () => request<CommunityStats>("/api/community/stats"),
};

// ===== 面试经验 =====
export const interviewApi = {
  submit: (body: InterviewSubmit) =>
    request<InterviewReport>("/api/interview/submit", {
      method: "POST",
      body: JSON.stringify(body),
    }),
  myReports: (params?: { page?: number; page_size?: number }) =>
    request<PaginatedResponse<InterviewReport>>(
      `/api/interview/my-reports${buildQuery((params as Record<string, string | undefined | null>) || {})}`,
    ),
  remove: (id: string) =>
    request<void>(`/api/interview/${id}`, { method: "DELETE" }),
  aggregate: (body: { company: string; position?: string }) =>
    request<InterviewAggregate>("/api/interview/aggregate", {
      method: "POST",
      body: JSON.stringify(body),
    }),
  stats: () => request<InterviewStats>("/api/interview/stats"),
  companies: (keyword: string = "") =>
    request<CompanyInfo[]>("/api/interview/companies", {
      method: "POST",
      body: JSON.stringify({ keyword }),
    }),
};